src/test/test_ptr_slow.c

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2001-2004, Roger Dingledine.
+ * Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+ * Copyright (c) 2007-2019, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+#include "orconfig.h"
+#include "core/or/or.h"
+#include "test/test.h"
+
+#include <stdint.h>
+#include <limits.h>
+
+/** Assert that <b>a</b> can be cast to void * and back. */
+static void
+assert_int_voidptr_roundtrip(int a)
+{
+  intptr_t ap = (intptr_t)a;
+  void *b = (void *)ap;
+  intptr_t c = (intptr_t)b;
+  void *d = (void *)c;
+
+  tt_assert(ap == c);
+  tt_assert(b == d);
```

Comment:
A smart optimiser may compile this function down to nothing.
(Has anyone checked what the current clang and gcc do?)

We can avoid most optimisations by placing each kind of cast in a function in another C file.
(Telling compilers not to optimise may not work: some compilers do some optimisations as a necessary part of translation.)

Reply:
Moved the cast operations to separate file in 0bc9ed9d38cb29783d000b6e22677ae16727976b.